<launch>
    <arg name="camera_name" default="camera"/>
    <arg name="3d_sensor" default="gemini_330_series"/>
    <arg name="camera1_prefix" default="01"/>
    <arg name="camera2_prefix" default="02"/>
    <arg name="camera3_prefix" default="03"/>
    <arg name="camera4_prefix" default="04"/>
    <!-- <arg name="camera1_usb_port" default="2-1"/>
    <arg name="camera2_usb_port" default="2-2"/>
    <arg name="camera3_usb_port" default="2-3"/>
    <arg name="camera4_usb_port" default="2-7"/>
    <arg name="camera5_usb_port" default="2-9"/> -->
    <!-- <arg name="camera1_serial" default=""/>
    <arg name="camera2_serial" default=""/>
    <arg name="camera3_serial" default=""/>
    <arg name="camera4_serial" default=""/>
    <arg name="camera5_serial" default=""/> -->
    <arg name="camera1_serial" default="CP728410000C"/>
    <arg name="camera2_serial" default="CP7284100083"/>
    <arg name="camera3_serial" default="CP72841000CX"/>
    <arg name="camera4_serial" default="CP7284100016"/>

    <arg name="camera1_usb_port" default=""/>
    <arg name="camera2_usb_port" default=""/>
    <arg name="camera3_usb_port" default=""/>
    <arg name="camera4_usb_port" default=""/>

    <arg name="device_num" default="4"/>
    <include file="$(find orbbec_camera)/launch/$(arg 3d_sensor).launch">
        <arg name="camera_name" value="$(arg camera_name)_$(arg camera1_prefix)"/>
        <arg name="usb_port" value="$(arg camera1_usb_port)"/>
        <arg name="serial_number" value="$(arg camera1_serial)"/>
        <arg name="device_num" value="$(arg device_num)"/>
    </include>

    <include file="$(find orbbec_camera)/launch/$(arg 3d_sensor).launch">
        <arg name="camera_name" value="$(arg camera_name)_$(arg camera2_prefix)"/>
        <arg name="usb_port" value="$(arg camera2_usb_port)"/>
        <arg name="serial_number" value="$(arg camera2_serial)"/>
        <arg name="device_num" value="$(arg device_num)"/>
    </include>

    <include file="$(find orbbec_camera)/launch/$(arg 3d_sensor).launch">
        <arg name="camera_name" value="$(arg camera_name)_$(arg camera3_prefix)"/>
        <arg name="usb_port" value="$(arg camera3_usb_port)"/>
        <arg name="serial_number" value="$(arg camera3_serial)"/>
        <arg name="device_num" value="$(arg device_num)"/>
    </include>

    <include file="$(find orbbec_camera)/launch/$(arg 3d_sensor).launch">
        <arg name="camera_name" value="$(arg camera_name)_$(arg camera4_prefix)"/>
        <arg name="usb_port" value="$(arg camera4_usb_port)"/>
        <arg name="serial_number" value="$(arg camera4_serial)"/>
        <arg name="device_num" value="$(arg device_num)"/>
    </include>


    <node pkg="tf" type="static_transform_publisher" name="camera_01_tf" args="0 0 0 0 0 0 base_link camera_01_link 20" />
    <node pkg="tf" type="static_transform_publisher" name="camera_02_tf" args="0 0.3 0 0 0 0 base_link camera_02_link 20" />
    <node pkg="tf" type="static_transform_publisher" name="camera_03_tf" args="0 0.6 0 0 0 0 base_link camera_03_link 20" />
    <node pkg="tf" type="static_transform_publisher" name="camera_04_tf" args="0 0.9 0 0 0 0 base_link camera_04_link 20" />
</launch>
